src/just_tls.erl

```diff
@@ -110,16 +103,25 @@ make_ssl_opts(Opts) ->
     {dummy_ref, SSLOpts} = format_opts_with_ref(Opts),
     SSLOpts.
 
+%% @doc Preprocess SSL options for tcp_to_tls calls
+-spec prepare_options(mongoose_tls:options()) -> [ssl:tls_option()].
+prepare_options(Opts) ->
+    Verify = verify_opt(Opts),
+    SNIOpts = sni_opts(Opts),
+    SSLOpts = maps:to_list(maps:with(ssl_option_keys(), Opts)),
+    [{verify, Verify}] ++ SNIOpts ++ SSLOpts.
+
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %% local functions
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
+-spec format_opts_with_ref(mongoose_tls:options()) -> {_, [ssl:tls_option()]}.
+format_opts_with_ref(#{prepare_options := SslOptions} = Opts) ->
```

Comment:
I don't like the fact that we accept both prepared and unprepared options here, we had such code in the past, and it was error-prone and confusing.